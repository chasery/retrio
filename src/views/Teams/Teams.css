.Teams__wrapper {
  display: flex;
  flex-direction: column;
}

.Teams__header {
  display: flex;
}
.Teams__header > h2 {
  flex-grow: 1;
}

.Teams__addTeam {
  align-self: center;
  margin-top: 1rem;
}
