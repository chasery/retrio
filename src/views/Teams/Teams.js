import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import TeamsApiService from '../../services/teams-api-service';
import Header from '../../components/Header/Header';
import TeamsList from '../../components/TeamsList/TeamsList';
import Error from '../../components/Error/Error';
import './Teams.css';

function Teams(props) {
  const [teams, setTeams] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    async function initState() {
      try {
        let apiCall = await TeamsApiService.getTeams();
        let res = await apiCall;

        setTeams(res);
      } catch (error) {
        setError(error.error);
      }
    }

    initState();
  }, []);

  return (
    <>
      <Header />
      <main role='main'>
        <section className='Teams'>
          <div className='Teams__wrapper'>
            <div className='Teams__header'>
              <h2>My Teams</h2>
              <Link to='/teams/add'>Add Team</Link>
            </div>
            {error ? <Error message={error} /> : <TeamsList teams={teams} />}
          </div>
        </section>
      </main>
    </>
  );
}

export default Teams;
