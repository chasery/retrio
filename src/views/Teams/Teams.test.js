import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Teams from './Teams';

describe('Teams component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<Teams />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
