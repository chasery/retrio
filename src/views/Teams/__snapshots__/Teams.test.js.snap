// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="Teams"
    >
      <div
        className="Teams__wrapper"
      >
        <div
          className="Teams__header"
        >
          <h2>
            My Teams
          </h2>
          <Link
            to="/teams/add"
          >
            Add Team
          </Link>
        </div>
        <TeamsList
          teams={Array []}
        />
      </div>
    </section>
  </main>
</Fragment>
`;
